code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -454,19 +453,18 @@
 	. = ..()
 	if(!.)
 		return .
-	for(var/obj/item/borg/cyborg_omnitool/engineering/omnitool_upgrade in cyborg.model.modules)
-		if(omnitool_upgrade.upgraded)
-			to_chat(user, span_warning("This unit is already equipped with an omnitool upgrade!"))
-			return FALSE
-	for(var/obj/item/borg/cyborg_omnitool/engineering/omnitool in cyborg.model.modules)
-		omnitool.upgrade_omnitool()
+	if(cyborg.model.toolbox.currently_upgraded)
+		to_chat(user, span_warning("This unit is already equipped with an omnitool upgrade!"))
+		return FALSE
+	cyborg.model.toolbox.currently_upgraded = TRUE
+	cyborg.model.toolbox.set_upgrade()
```

Comment:
It'd be nicer to pass TRUE/FALSE in as an argument than set the variable externally before calling the proc, I think.